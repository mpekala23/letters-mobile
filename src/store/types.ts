import { UserState } from "./User/UserTypes";
import { NotifState } from "./Notif/NotifTypes";
import { ContactState } from "./Contact/ContactTypes";

export interface AppState {
  user: UserState;
  notif: NotifState;
  contact: ContactState;
}
