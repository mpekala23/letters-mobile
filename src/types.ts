// TODO: make this typing better
// probably want a specific type for state abbrevs
export interface Facility {
  name: string;
  type: PrisonTypes;
  address: string;
  city: string;
  state: string;
  postal: string;
  phone: string;
}

export enum ProfilePicTypes {
  Topbar = 'Topbar',
  Contact = 'Contact',
  SingleContact = 'SingleContact',
}

export enum Storage {
  RememberToken = 'Ameelio-Token',
}

export enum PrisonTypes {
  State = 'State',
  Federal = 'Federal',
  Jail = 'County',
}

export enum SupportFAQTypes {
  DeleteLetter = 'DeleteLetter',
  NotArrived = 'NotArrived',
  WrongReturnAddress = 'WrongReturnAddress',
  WrongMailingAddress = 'WrongMailingAddress',
  TrackingNumber = 'TrackingNumber',
  TrackingError = 'TrackingError',
  TalkToAmeelio = 'TalkToAmeelio',
}

export enum LetterStatus {
  Draft = 'Draft',
  Created = 'Created',
  Mailed = 'Mailed',
  InTransit = 'In Transit',
  InLocalArea = 'In Local Area',
  OutForDelivery = 'Out for Delivery',
  Delivered = 'Delivered',
}

export enum LetterTypes {
  Postcard = 'postcard',
  Letter = 'letter',
}

export enum DeliveryReportTypes {
  received = 'received',
  unsure = 'unsure',
  notYetReceived = 'notYetReceived',
  haveNotAsked = 'haveNotAsked',
  haveNotReceived = 'haveNotReceived',
}

export interface Letter {
  type: LetterTypes;
  status: LetterStatus;
  isDraft: boolean;
  recipientId: number;
  content: string;
  photo?: Photo;
  letterId?: number; // TODO: Once we have more info on this field and lob, use this more
  expectedDeliveryDate?: Date;
  dateCreated?: Date;
  trackingEvents?: LetterTrackingEvent[];
}

export interface LetterTrackingEvent {
  id: number;
  name: string;
  location: ZipcodeInfo;
  date: Date;
}

export type TopbarRouteAction = {
  enabled: boolean;
  text: string;
  action: () => void | Promise<void>;
  blocking?: boolean;
};

export interface ZipcodeInfo {
  zip: string;
  city: string;
  state: string;
  lat?: number;
  long?: number;
}

export interface Photo {
  type?: 'image' | string;
  width?: number;
  height?: number;
  uri: string;
}
