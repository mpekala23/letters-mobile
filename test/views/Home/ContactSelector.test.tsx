import * as React from 'react';
import { ContactSelectorScreen } from '@views';
import { render, toJSON, fireEvent } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { LetterTypes, LetterStatus } from 'types';

const mockStore = configureStore([]);

const setup = (contactsOverrides = [], lettersOverrides = []) => {
  const navigation = { navigate: jest.fn(), addListener: jest.fn() };
  const contacts = Object.assign(
    [
      {
        state: 'Minnesota',
        firstName: 'First',
        lastName: 'Last',
        inmateNumber: '6',
        relationship: 'Brother',
      },
    ],
    contactsOverrides
  );

  const letters = {
    type: LetterTypes.Postcard,
    status: LetterStatus.Printed,
    isDraft: true,
    recipientId: 8,
    content: "I'm trying out this new service called Ameelio...",
    ...lettersOverrides,
  };

  const initialContactState = {
    adding: {},
    existing: contacts,
  };

  const initialLetterState = {
    composing: {},
    existing: letters,
  };

  const store = mockStore({
    contact: initialContactState,
    letter: initialLetterState,
  });

  const StoreProvider = ({ children }: { children: JSX.Element }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return {
    navigation,
    store,
    ...render(
      <ContactSelectorScreen
        navigation={navigation}
        contactState={initialContactState}
        letterState={initialLetterState}
      />,
      {
        wrapper: StoreProvider,
      }
    ),
  };
};

describe('Contact Selector Screen', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should load values for contact cards from the redux store', () => {
    const { getByText } = setup([
      {
        state: 'Minnesota',
        firstName: 'First Contact',
        lastName: 'Contact Last Name',
        inmateNumber: '12345678',
        relationship: 'Brother',
      },
      {
        state: 'Minnesota',
        firstName: 'Second Contact',
        lastName: 'Second Contact Last Name',
        inmateNumber: '12345678',
        relationship: 'Brother',
      },
    ]);
    expect(getByText('First Contact').props.children).toBe('First Contact');
    expect(getByText('Second Contact').props.children).toBe('Second Contact');
  });

  it('should navigate to contact info screen when the plus button is pressed', () => {
    const { navigation, getByTestId } = setup();
    fireEvent.press(getByTestId('addContact'));
    expect(navigation.navigate).toHaveBeenCalledWith('ContactInfo', {
      addFromSelector: true,
    });
  });
});
