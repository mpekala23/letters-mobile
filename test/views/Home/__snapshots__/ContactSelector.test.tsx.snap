// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Selector Screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#EDEDED",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": 16,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins-Bold",
          },
          Object {
            "color": "#6D6D6D",
            "fontSize": 20,
            "paddingBottom": 16,
          },
        ]
      }
    >
      Your loved ones
    </Text>
    <ScrollView
      data={
        Array [
          Object {
            "firstName": "First",
            "inmateNumber": "6",
            "lastName": "Last",
            "relationship": "Brother",
            "state": "Minnesota",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      refreshControl={
        <RefreshControl
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        style={null}
      >
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 6,
                "marginVertical": 6,
                "padding": 15,
                "width": "100%",
              },
              Object {
                "elevation": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 2.62,
              },
              undefined,
            ]
          }
          testID="ContactSelectorCard"
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "paddingTop": 12,
                  },
                ]
              }
            >
              <View
                pointerEvents="none"
              >
                <TouchableOpacity
                  activeOpacity={0.2}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#f6993f",
                      "borderRadius": 40,
                      "height": 80,
                      "justifyContent": "center",
                      "overflow": "hidden",
                      "width": 80,
                    }
                  }
                  testID="profilePicture"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Poppins-Regular",
                        },
                        Object {
                          "color": "white",
                          "fontSize": 22,
                          "fontWeight": "500",
                        },
                      ]
                    }
                  >
                    FL
                  </Text>
                </TouchableOpacity>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "paddingLeft": 18,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 20,
                    },
                  ]
                }
              >
                First
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                    Object {
                      "color": "#6D6D6D",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000",
                    }
                  }
                >
                  üíå
                </Text>
                 
                received
                :
                 
                0
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                    Object {
                      "color": "#6D6D6D",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000",
                    }
                  }
                >
                  üìÜ
                </Text>
                 
                last heard from you
                :
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Poppins-Regular",
                    },
                    Object {
                      "color": "#6D6D6D",
                      "paddingBottom": 4,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000",
                    }
                  }
                >
                  ‚úàÔ∏è
                </Text>
                 
                letters traveled
                :
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
    </ScrollView>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FC7272",
          "borderRadius": 50,
          "bottom": 0,
          "height": 65,
          "marginBottom": 24,
          "marginRight": 12,
          "paddingLeft": 2,
          "position": "absolute",
          "right": 0,
          "width": 65,
        }
      }
      testID="addContact"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={39}
        bbWidth={39}
        fill="none"
        focusable={false}
        height={39}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "marginRight": 2,
              "marginTop": 13,
            },
            Object {
              "flex": 0,
              "height": 39,
              "width": 39,
            },
          ]
        }
        vbHeight={39}
        vbWidth={39}
        width={39}
        xml="<svg width=\\"39\\" height=\\"39\\" viewBox=\\"0 0 39 39\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<rect y=\\"16\\" width=\\"39\\" height=\\"7\\" rx=\\"3.5\\" fill=\\"white\\"/>
<rect x=\\"23\\" width=\\"39\\" height=\\"7\\" rx=\\"3.5\\" transform=\\"rotate(90 23 0)\\" fill=\\"white\\"/>
</svg>"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGRect
            fill={4294967295}
            height={7}
            propList={
              Array [
                "fill",
              ]
            }
            rx={3.5}
            width={39}
            x={0}
            y={16}
          />
          <RNSVGRect
            fill={4294967295}
            height={7}
            matrix={
              Array [
                6.123233995736766e-17,
                1,
                -1,
                6.123233995736766e-17,
                23,
                -23,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rx={3.5}
            width={39}
            x={23}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </TouchableOpacity>
  </View>
</View>
`;
